src/main/java/com/thealgorithms/sorts/CountingSort.java

```diff
@@ -0,0 +1,54 @@
+package com.thealgorithms.sorts;
+
+import java.util.Arrays;
+
+/**
+ * A standard implementation of the Counting Sort algorithm for integer arrays.
+ * This implementation has a time complexity of O(n + k), where n is the number
+ * of elements in the input array and k is the range of the input.
+ * It works only with integer arrays.
+ *
+ * The space complexity is O(k), where k is the range of the input integers.
+ *
+ * Note: This implementation does not handle negative integers as it
+ * calculates the range based on the minimum and maximum values of the array.
```

Comment:
I think your implementation indeed works for negative numbers.

By the way: to inform the user by throwing an exception for example.

Reply:
Yes, fixed this. Thanks.